internal/testrunner/runners/system/servicedeployer/terraform.go

```diff
@@ -117,10 +158,28 @@ func (tsd TerraformServiceDeployer) SetUp(inCtxt ServiceContext) (DeployedServic
 	}
 
 	outCtxt.Agent.Host.NamePrefix = "docker-fleet-agent"
+
+	err = addTerraformOutputs(outCtxt)
+	if err != nil {
+		return nil, fmt.Errorf("could not handle terraform output %w", err)
+	}
 	service.ctxt = outCtxt
 	return &service, nil
 }
 
+func createOutputDir(runId string) (string, error) {
+	buildDir, err := builder.BuildDirectory()
+	if err != nil {
+		return "", fmt.Errorf("locating build directory failed %w", err)
+	}
+
+	od := filepath.Join(buildDir + runId)
+	if err := os.MkdirAll(od, 0755); err != nil {
+		return "", fmt.Errorf("failed to create output directory: %w", err)
+	}
+	return od, nil
```

Comment:
nit. instead of concatenating the strings in the filepath.Join, let the method to concatenate both to ensure everything is ok.

And I would use complete words for the variables:

```suggestion
	outputDir := filepath.Join(buildDir, runId)
	if err := os.MkdirAll(outputDir, 0755); err != nil {
		return "", fmt.Errorf("failed to create output directory: %w", err)
	}
	return outputDir, nil
```